lib/pure/asynchttpserver.nim

```diff
@@ -404,6 +404,7 @@ proc acceptRequest*(server: AsyncHttpServer,
 proc serve*(server: AsyncHttpServer, port: Port,
             callback: proc (request: Request): Future[void] {.closure, gcsafe.},
             address = "";
+            domain: Domain = AF_INET;
             assumedDescriptorsPerRequest = -1) {.async.} =
```

Comment:
Maybe you can make a new overload to maintain backward compatibility.

```nim
proc serve*(server: AsyncHttpServer, port: Port,
            callback: proc (request: Request): Future[void] {.closure, gcsafe.},
            address: string;
            domain: Domain;
            assumedDescriptorsPerRequest = -1) {.async.} =
```

Reply:
Moved domain to the end.